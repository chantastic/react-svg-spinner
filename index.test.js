import React from "react";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer"
import Spinner from "."

Enzyme.configure({ adapter: new Adapter() });

test("Spinner renders", () => {
  const spinner = shallow(<Spinner />)
})

test("no prop snapshot", () => {
  const component = renderer.create(<Spinner />)

	let tree = component.toJSON()

	expect(tree).toMatchSnapshot();
})
