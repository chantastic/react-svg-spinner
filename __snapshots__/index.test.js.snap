// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no prop snapshot 1`] = `
<svg
  aria-labelledby="title desc"
  className="__react-svg-spinner_circle"
  height="1em"
  role="img"
  style={
    Object {
      "animationDuration": "750ms",
    }
  }
  viewBox="0 0 32 32"
  width="1em"
>
  <title
    id="title"
  >
    Circle loading spinner
  </title>
  <desc
    id="desc"
  >
    Image of a partial circle indicating "loading."
  </desc>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "
      .__react-svg-spinner_circle{
          transition-property: transform;
          animation-name: __react-svg-spinner_infinite-spin;
          animation-iteration-count: infinite;
          animation-timing-function: linear;
      }
      @keyframes __react-svg-spinner_infinite-spin {
          from {transform: rotate(0deg)}
          to {transform: rotate(360deg)}
      }
    ",
      }
    }
  />
  <circle
    cx={16}
    cy={16}
    fill="none"
    r={12}
    role="presentation"
    stroke="rgba(0,0,0,0.4)"
    strokeDasharray={43.982297150257104}
    strokeLinecap="round"
    strokeWidth={4}
  />
</svg>
`;
